File: src/main/java/com/trade/securities/adapter/in/web/MarketDataController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.GetMarketDataUseCase;
import com.trade.securities.domain.MarketData;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/market-data")
public class MarketDataController {

    private final GetMarketDataUseCase getMarketDataUseCase;

    @GetMapping
    public ResponseEntity<List<MarketData>> getMarketData(
            @org.springframework.web.bind.annotation.RequestParam(name = "date", defaultValue = "20241227") String date,
            @org.springframework.web.bind.annotation.RequestParam(name = "type", defaultValue = "KOSPI") String type) {
        return ResponseEntity.ok(getMarketDataUseCase.getMarketData(date, type));
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/MarketDataController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.GetMarketDataUseCase
import com.trade.securities.domain.MarketData
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.DefaultValue
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/market-data")
@Produces(MediaType.APPLICATION_JSON)
class MarketDataController(
    private val getMarketDataUseCase: GetMarketDataUseCase
) {

    @GET
    fun getMarketData(
        @QueryParam("date") @DefaultValue("20241227") date: String,
        @QueryParam("type") @DefaultValue("KOSPI") type: String
    ): List<MarketData> {
        return getMarketDataUseCase.getMarketData(date, type)
    }
}
